examples/models/llama2/evaluate/eager_eval.py

```diff
@@ -77,10 +79,21 @@ def tok_decode(self, tokens):
 
     def _model_call(self, inps):
         if self._use_kv_cache:
-            pos_tensor = torch.tensor([0], dtype=torch.int64, device=self.device)
-            # Batch process the whole sequence.
-            logits = self._model(inps[:, : self._max_seq_length], pos_tensor)
-            return logits
+            if not self._dynamic_shape:
+                # graph module exported without dynamic shape won't work with a different shape.
+                # And we have to do single token prefill here.
+                result_logits = []
+                for pos in range(self._max_seq_length):
```

Comment:
inps's length is not guaranteed to be `_max_seq_length`. It is safer to loop through `range(inps.shape[-1])`.